### DISCUSSION TITLE
Implement a retry logic for 429 and 500 errors?

### DISCUSSION THREAD

**Discussion by createreadupdate** [openai-php/client#156] [Ideas]
Could be put into HttpTransporter with an additional public method like ->setMaxRetries and a default of 1 to have no effect on the library as it is.

Some increasing sleep between the retries would be good too.

What do you think?

**Comment by ondraasek** [comment]
+1

**Comment by hschimpf** [comment]
You should implement the retry mechanism on your side:

A simple example could be as follows:

```php
$tries = 5;
$sleep = 0;
do {
    try {
        sleep($sleep);

        $response = $client::chat()->create($parameters);
    } catch (Exception $e) {
        if (--$tries === 0) {
            throw $e;
        }

        $response = null;
        $sleep++;
    }
} while ($response === null);
```
